### PR TITLE
 [IMP] web,*: better classes handling in view archs

### PR DESCRIPTION
**Pull request by Polymorphe57**
**Before**
- The optional `class` attribute set on the root node of a view arch
  is ignored, except for the kanban view which has a custom
  way of using it.
- The optional `js_class` attribute set on the root node of a view arch
  does not have any impact on the class names passed to its controller.

**After**
The content of the optional attribute `class` set on the root node of an
arch like in

```xml
<list class="o_custom_class">
    ...
</list>
```

as well as an additionnal class derived [1] from the value of the
`js_class` attribute set on the root node of an arch like in

```xml
<list js_class="extended_list">
    ...
</list>
```

will both be found in the prop `className` of any view controller.

[1] a js_class value of `xyz` yields to the class `o_xyz_view`

**Additional note**
The kanban view was already appending the root node class attribute
to its renderer element. This is no longer the case and some styling
rules has been adapted.

---

Related Enterprise PR: odoo/enterprise#37265

### CODE REVIEW

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by brboi** [comment]
@aab-odoo @Polymorphe57 
I got a fix for the fact the staging failed ;-)

I am preparing the branch and force-pushing it in a moment :-)

File: addons/hr/static/src/scss/hr.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.o_kanban_dashboard.o_hr_department_kanban {
+.o_kanban_dashboard.o_hr_department_kanban .o_kanban_renderer {
```
Comment (Polymorphe57) [inline addons/hr/static/src/scss/hr.scss:1]: what is the purpose of oe_background_grey?

Comment (brboi) [reply addons/hr/static/src/scss/hr.scss:1]: no idea 😅

File: odoo/addons/base/static/src/scss/res_partner.scss [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.o_kanban_view .o_res_partner_kanban {
+.o_kanban_view.o_res_partner_kanban .o_kanban_renderer {
```
Comment (Polymorphe57) [inline odoo/addons/base/static/src/scss/res_partner.scss:1]: nope: o_res_partner_kanban is used in a kanban-box

File: addons/survey/static/src/scss/survey_survey_views.scss [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // KANBAN VIEW
-.o_survey_survey_view_kanban {
+.o_survey_survey_view_kanban .o_kanban_renderer {
```
Comment (Polymorphe57) [inline addons/survey/static/src/scss/survey_survey_views.scss:2]: maybe remove class (survey_survey lol) and use js_class instead (survey_view_kanban)

--> in other simple cases: do that too?

File: addons/stock/views/stock_picking_type_views.xml [hunk 4]
```diff
@@ -129,7 +129,7 @@
         <field name="name">stock.picking.type.kanban</field>
         <field name="model">stock.picking.type</field>
         <field name="arch" type="xml">
-            <kanban class="oe_background_grey o_kanban_dashboard o_emphasize_colors o_stock_kanban" create="0" group_create="false">
+            <kanban class="o_kanban_dashboard o_emphasize_colors o_stock_kanban" create="0" group_create="false">
```
Comment (Polymorphe57) [inline addons/stock/views/stock_picking_type_views.xml:132]: o_stock_kanban seems to be unused

File: addons/website_slides/views/slide_channel_views.xml [hunk 5]
```diff
@@ -236,7 +236,7 @@
             <field name="name">slide.channel.view.kanban</field>
             <field name="model">slide.channel</field>
             <field name="arch" type="xml">
-                <kanban string="eLearning Overview" class="o_emphasize_colors o_kanban_dashboard o_slide_kanban breadcrumb_item active" edit="false" sample="1">
+                <kanban string="eLearning Overview" class="o_emphasize_colors o_kanban_dashboard o_slide_kanban" edit="false" sample="1">
```
Comment (Polymorphe57) [inline addons/website_slides/views/slide_channel_views.xml:239]: breadcrumb_item active? why?

Comment (brboi) [reply addons/website_slides/views/slide_channel_views.xml:239]: no valid reason to leave this

File: addons/project/views/project_views.xml [hunk 6]
```diff
@@ -786,7 +786,7 @@
             <field name="model">project.project</field>
             <field name="arch" type="xml">
                 <kanban
-                    class="oe_background_grey o_kanban_dashboard o_project_kanban o_emphasize_colors"
+                    class="o_kanban_dashboard o_project_kanban o_emphasize_colors"
```
Comment (Polymorphe57) [inline addons/project/views/project_views.xml:789]: maybe use project_kanban as js_class and remove o_project_kanban from class?

File: addons/project/static/src/views/project_update_kanban/project_update_kanban.scss [hunk 7]
```diff
@@ -1,4 +1,4 @@
-.o_pupdate_kanban {
+.o_pupdate_kanban .o_kanban_renderer {
```
Comment (Polymorphe57) [inline addons/project/static/src/views/project_update_kanban/project_update_kanban.scss:1]: use js_class o_pupdate_kanban?

File: addons/project/static/src/views/project_task_kanban/project_task_kanban_view.scss [hunk 8]
```diff
@@ -1,4 +1,4 @@
-.o_kanban_project_tasks {
+.o_kanban_project_tasks .o_kanban_renderer {
```
Comment (Polymorphe57) [inline addons/project/static/src/views/project_task_kanban/project_task_kanban_view.scss:1]: useless diff

File: addons/project/static/src/scss/project_widgets.scss [hunk 9]
```diff
@@ -37,7 +37,7 @@
     cursor: pointer;
 }
 
-.o_kanban_project_tasks .o_field_many2manytags, .o_kanban_tags{
+.o_kanban_project_tasks .o_kanban_renderer .o_field_many2manytags, .o_kanban_tags{
```
Comment (Polymorphe57) [inline addons/project/static/src/scss/project_widgets.scss:40]: useless diff

File: addons/hr_expense/static/src/views/kanban.xml [hunk 10]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <templates xml:space="preserve">
     <t t-name="hr_expense.KanbanRenderer" t-inherit="web.KanbanRenderer" t-inherit-mode="primary" owl="1">
+        <xpath expr="//div[hasclass('o_kanban_renderer')]" position="attributes">
+            <attribute name="class">o_kanban_renderer o_renderer d-flex position-relative h-auto</attribute>
+        </xpath>
```
Comment (aab-odoo) [inline addons/hr_expense/static/src/views/kanban.xml:6]: use add?

**Review by aab-odoo — APPROVED** [review]

**Review by aab-odoo — APPROVED** [review]
robodoo rebase-ff r+

**Pull request closed without merging**